internal/upload/ftp.go

```diff
@@ -268,71 +137,29 @@ func (agent *FTPTransferAgent) GetReturnFiles() ([]File, error) {
 	return agent.readFiles(agent.cfg.Paths.Return)
 }
 
-func (agent *FTPTransferAgent) readFiles(path string) ([]File, error) {
-	agent.mu.Lock()
-	defer agent.mu.Unlock()
+func (agent *FTPTransferAgent) readFiles(dir string) ([]File, error) {
+	var files []File
 
-	conn, err := agent.connection()
+	filenames, err := agent.client.ListFiles(dir)
 	if err != nil {
 		return nil, err
 	}
 
-	// move into inbound directory and set a trigger to undo
-	wd, err := conn.CurrentDir()
-	if err != nil {
-		return nil, err
-	}
-	defer func(path string) {
-		// Return to our previous directory when initially called
-		if err := conn.ChangeDir(wd); err != nil {
-			agent.logger.LogErrorf("FTP: problem with readFiles: %v", err)
+	for i := range filenames {
```

Comment:
Any reason for using the index and not the value? I find using the slice value cleaner and less cluttered.

Reply:
`for _, v := range list` is a pattern [with some known issues](https://github.com/golang/go/discussions/56010) so I've always tried to avoid coding in a way which could have unexpected bugs. The Go language is working on fixing the issue I referenced, but the muscle memory is in my hands already.

Aside from that issue the difference between `for i := range list` and `for _, v := range list` seems be a discussion about personal preference and style anyway, which I don't have much interest in debating. I have my preference and others have theirs, but a project should use a consistent style throughout.